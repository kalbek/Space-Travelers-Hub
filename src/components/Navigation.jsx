import { NavLink } from 'react-router-dom';

const Navigation = () => (
  <>
    <nav>
      <div className="nav-container">
        <div>
          <div className="flex">
            <div className="logo">Logo</div>
            <div className="title">Space Travellers&apos; Hub</div>
          </div>
          <div className="menus flex gap1">
            <NavLink
              to="/rockets"
              activeClassName="selected"
              style={{ textDecoration: 'none' }}
            >
              <div className="menu ptr">Rockets</div>
            </NavLink>
            <NavLink
              to="/missions"
              activeClassName="selected"
              style={{ textDecoration: 'none' }}
            >
              <div className="menu ptr">Missions</div>
            </NavLink>
            |
            <NavLink
              to="/profile"
              activeClassName="selected"
              style={{ textDecoration: 'none' }}
            >
              <div className="menu ptr">My Profile</div>
            </NavLink>
          </div>
        </div>
      </div>
    </nav>
  </>
);

export default Navigation;
