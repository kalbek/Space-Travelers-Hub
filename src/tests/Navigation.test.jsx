import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import Navigation from '../components/Navigation';

describe('Navigation component', () => {
  test('renders Navigation component', () => {
    const { container } = render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  test('checks if the text "Space Travellers\' Hub" is present', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
    );
    const linkElement = getByText(/Space Travellers' Hub/i);
    expect(linkElement).toBeInTheDocument();
  });
});
