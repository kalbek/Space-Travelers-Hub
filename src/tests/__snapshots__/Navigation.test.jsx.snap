// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component renders Navigation component 1`] = `
<div>
  <nav>
    <div
      class="nav-container"
    >
      <div>
        <div
          class="flex"
        >
          <div
            class="logo"
          >
            Logo
          </div>
          <div
            class="title"
          >
            Space Travellers' Hub
          </div>
        </div>
        <div
          class="menus flex gap1"
        >
          <a
            activeclassname="selected"
            class=""
            href="/rockets"
            style="text-decoration: none;"
          >
            <div
              class="menu ptr"
            >
              Rockets
            </div>
          </a>
          <a
            activeclassname="selected"
            class=""
            href="/missions"
            style="text-decoration: none;"
          >
            <div
              class="menu ptr"
            >
              Missions
            </div>
          </a>
          |
          <a
            activeclassname="selected"
            class=""
            href="/profile"
            style="text-decoration: none;"
          >
            <div
              class="menu ptr"
            >
              My Profile
            </div>
          </a>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
